src/polynomial/mod.rs

```diff
@@ -55,21 +62,51 @@ impl<F: PrimeField> Expression<F> {
         }
     }
 
+    // return the number of challenges in expression
+    pub fn num_challenges(&self) -> usize {
+        let mut set = HashSet::<(i32, usize, usize)>::new();
+        self._num_challenges(&mut set);
+        set.len()
+    }
+
+    pub fn _num_challenges(&self, set: &mut HashSet<(i32, usize, usize)>) {
```

Comment:
Why you start pub fn with underscore? Please add rustdoc and explain a little a sense of this fn

Reply:
it should be internal function. forget to remove it.